src/header/dib_header.rs

```diff
@@ -127,7 +128,19 @@ pub enum HeaderType {
 }
 
 impl HeaderType {
-    fn is_at_least(self, header_type: HeaderType) -> bool {
-        self >= header_type
+    const fn is_at_least(self, header_type: HeaderType) -> bool {
+        matches!(
+            (self, header_type),
+            (Self::V5, Self::V5)
+                | (Self::V5, Self::V4)
+                | (Self::V5, Self::V3)
+                | (Self::V5, Self::Info)
+                | (Self::V4, Self::V4)
+                | (Self::V4, Self::V3)
+                | (Self::V4, Self::Info)
+                | (Self::V3, Self::V3)
+                | (Self::V3, Self::Info)
+                | (Self::Info, Self::Info)
+        )
```

Comment:
I think this should also work:
```suggestion
        self as u8 >= header_type as u8
```